pkg/fetcher/definition.go

```diff
@@ -22,33 +23,71 @@ type Definition struct {
 	reader client.Reader
 }
 
+type uniqueTestDefinitions map[types.UID]v1alpha1.TestDefinition
+
 func (s *Definition) FindMatching(suite v1alpha1.ClusterTestSuite) ([]v1alpha1.TestDefinition, error) {
 	ctx := context.TODO()
-	if len(suite.Spec.Selectors.MatchNames) > 0 {
-		return s.findByNames(ctx, suite)
+	acc := make(uniqueTestDefinitions)
+
+	err := s.findByNames(ctx, suite, acc)
+	if err != nil {
+		return nil, err
 	}
-	// TODO(aszecowka) later so far we return all test definitions for all namespaces (https://github.com/kyma-incubator/octopus/issues/7)
+
+	err = s.findByLabelExpressions(ctx, suite, acc)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(acc) > 0 {
+		return acc.getValues(), nil
+	}
+
 	var list v1alpha1.TestDefinitionList
 	if err := s.reader.List(ctx, &client.ListOptions{Namespace: ""}, &list); err != nil {
 		return nil, errors.Wrap(err, "while listing test definitions")
 	}
 	return list.Items, nil
 }
 
-func (s *Definition) findByNames(ctx context.Context, suite v1alpha1.ClusterTestSuite) ([]v1alpha1.TestDefinition, error) {
-	var list []v1alpha1.TestDefinition
+func (s *Definition) findByNames(ctx context.Context, suite v1alpha1.ClusterTestSuite, acc uniqueTestDefinitions) error {
```

Comment:
why did you move the result to the arg list, for me it is less readable. The name 'findByNames' finds objects and what next? the same comment could be applied to findByLabelExpressions. The code which uses this method is not clear (lines 32 - 37)